pym/bob/cmds/build.py

```diff
@@ -946,8 +950,11 @@ def _cookPackageStep(self, packageStep, checkoutOnly, depth):
                         packageHash = hashWorkspace(packageStep)
                         workspaceChanged = True
                         wasDownloaded = True
-                    elif self.__forcedDownload:
-                        raise BuildError("Downloading artifact failed")
+                    elif self.__forcedDownload > 0:
+                        if self.__downloadDepth == 0 and self.__forcedDownload == 2:
+                            self.__downloadDepth = 1
```

Comment:
No, you must not modify the settings. The LocalBuilder object is used multiple times if multiple packages are passed on the command line. My Idea was to define a `__downloadDepthForce` (which is >= `__downloadDepth`) and you just do:

```python
elif depth >= self.__downloadDepthForce:
    raise BuildError("Downloading artifact failed")
```